// abz-components-library/fesm5/abz-components-library.js
import { Component, Input, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var ComponentsLibraryComponent = /** @class */ (function () {
    function ComponentsLibraryComponent() {
        this.name = 'default';
    }
    /**
     * @return {?}
     */
    ComponentsLibraryComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    ComponentsLibraryComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-components-library',
                    template: "\n    <p>\n      components-library works!\n    </p>\n  ",
                    styles: []
                },] },
    ];
    ComponentsLibraryComponent.ctorParameters = function () { return []; };
    ComponentsLibraryComponent.propDecorators = {
        name: [{ type: Input }]
    };
    return ComponentsLibraryComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var ComponentsLibraryModule = /** @class */ (function () {
    function ComponentsLibraryModule() {
    }
    ComponentsLibraryModule.decorators = [
        { type: NgModule, args: [{
                    imports: [],
                    declarations: [ComponentsLibraryComponent],
                    exports: [ComponentsLibraryComponent]
                },] },
    ];
    return ComponentsLibraryModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { ComponentsLibraryComponent, ComponentsLibraryModule };
