// abz-components-library/fesm5/abz-components-library.js
import { Component, Input, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * Examples:  <example-url>https://stackblitz.com/edit/angular-sw5vxm?embed=1&file=src/app/app.module.ts</example-url>
 */
var LibraryButtonComponent = /** @class */ (function () {
    function LibraryButtonComponent() {
        this.name = 'default';
    }
    /**
     * @return {?}
     */
    LibraryButtonComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    LibraryButtonComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-button',
                    template: "<button type=\"button\">Click Me!</button>\n",
                    styles: []
                },] },
    ];
    LibraryButtonComponent.ctorParameters = function () { return []; };
    LibraryButtonComponent.propDecorators = {
        name: [{ type: Input }]
    };
    return LibraryButtonComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * Examples:  <example-url>https://stackblitz.com/edit/angular-sw5vxm?embed=1&file=src/app/app.module.ts</example-url>
 */
var LibraryCheckboxComponent = /** @class */ (function () {
    function LibraryCheckboxComponent() {
        this.name = 'default';
    }
    /**
     * @return {?}
     */
    LibraryCheckboxComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    LibraryCheckboxComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-checkbox',
                    template: "<input type=\"checkbox\" name=\"vehicle1\" value=\"Bike\"> I have a bike<br>\n<input type=\"checkbox\" name=\"vehicle2\" value=\"Car\"> I have a car<br>\n<input type=\"checkbox\" name=\"vehicle3\" value=\"Boat\" checked> I have a boat<br><br>\n",
                    styles: []
                },] },
    ];
    LibraryCheckboxComponent.ctorParameters = function () { return []; };
    LibraryCheckboxComponent.propDecorators = {
        name: [{ type: Input }]
    };
    return LibraryCheckboxComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var ComponentsLibraryModule = /** @class */ (function () {
    function ComponentsLibraryModule() {
    }
    ComponentsLibraryModule.decorators = [
        { type: NgModule, args: [{
                    imports: [],
                    declarations: [LibraryButtonComponent, LibraryCheckboxComponent],
                    exports: [LibraryButtonComponent, LibraryCheckboxComponent]
                },] },
    ];
    return ComponentsLibraryModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { LibraryButtonComponent, ComponentsLibraryModule, LibraryCheckboxComponent as ɵa };
