// abz-components-library/bundles/abz-components-library.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('abz-components-library', ['exports', '@angular/core'], factory) :
    (factory((global['abz-components-library'] = {}),global.ng.core));
}(this, (function (exports,core) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /**
     * Examples:  <example-url>https://stackblitz.com/edit/angular-sw5vxm?embed=1&file=src/app/app.module.ts</example-url>
     */
    var LibraryButtonComponent = (function () {
        function LibraryButtonComponent() {
            this.name = 'default';
        }
        /**
         * @return {?}
         */
        LibraryButtonComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        LibraryButtonComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'lib-button',
                        template: "<button type=\"button\">Click Me!</button>\n",
                        styles: []
                    },] },
        ];
        LibraryButtonComponent.ctorParameters = function () { return []; };
        LibraryButtonComponent.propDecorators = {
            name: [{ type: core.Input }]
        };
        return LibraryButtonComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /**
     * Examples:  <example-url>https://stackblitz.com/edit/angular-sw5vxm?embed=1&file=src/app/app.module.ts</example-url>
     */
    var LibraryCheckboxComponent = (function () {
        function LibraryCheckboxComponent() {
            this.name = 'default';
        }
        /**
         * @return {?}
         */
        LibraryCheckboxComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        LibraryCheckboxComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'lib-checkbox',
                        template: "<input type=\"checkbox\" name=\"vehicle1\" value=\"Bike\"> I have a bike<br>\n<input type=\"checkbox\" name=\"vehicle2\" value=\"Car\"> I have a car<br>\n<input type=\"checkbox\" name=\"vehicle3\" value=\"Boat\" checked> I have a boat<br><br>\n",
                        styles: []
                    },] },
        ];
        LibraryCheckboxComponent.ctorParameters = function () { return []; };
        LibraryCheckboxComponent.propDecorators = {
            name: [{ type: core.Input }]
        };
        return LibraryCheckboxComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var ComponentsLibraryModule = (function () {
        function ComponentsLibraryModule() {
        }
        ComponentsLibraryModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [],
                        declarations: [LibraryButtonComponent, LibraryCheckboxComponent],
                        exports: [LibraryButtonComponent, LibraryCheckboxComponent]
                    },] },
        ];
        return ComponentsLibraryModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    exports.LibraryButtonComponent = LibraryButtonComponent;
    exports.ComponentsLibraryModule = ComponentsLibraryModule;
    exports.ɵa = LibraryCheckboxComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
