// abz-components-library/esm5/lib/components-library.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var ComponentsLibraryComponent = /** @class */ (function () {
    function ComponentsLibraryComponent() {
        this.name = 'default';
    }
    /**
     * @return {?}
     */
    ComponentsLibraryComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    ComponentsLibraryComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-components-library',
                    template: "\n    <p>\n      components-library works!\n    </p>\n  ",
                    styles: []
                },] },
    ];
    ComponentsLibraryComponent.ctorParameters = function () { return []; };
    ComponentsLibraryComponent.propDecorators = {
        name: [{ type: Input }]
    };
    return ComponentsLibraryComponent;
}());
export { ComponentsLibraryComponent };
if (false) {
    /** @type {?} */
    ComponentsLibraryComponent.prototype.name;
}
