// abz-components-library/esm5/lib/library-checkbox.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
/**
 * Examples:  <example-url>https://stackblitz.com/edit/angular-sw5vxm?embed=1&file=src/app/app.module.ts</example-url>
 */
var LibraryCheckboxComponent = /** @class */ (function () {
    function LibraryCheckboxComponent() {
        this.name = 'default';
    }
    /**
     * @return {?}
     */
    LibraryCheckboxComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    LibraryCheckboxComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-checkbox',
                    template: "<input type=\"checkbox\" name=\"vehicle1\" value=\"Bike\"> I have a bike<br>\n<input type=\"checkbox\" name=\"vehicle2\" value=\"Car\"> I have a car<br>\n<input type=\"checkbox\" name=\"vehicle3\" value=\"Boat\" checked> I have a boat<br><br>\n",
                    styles: []
                },] },
    ];
    LibraryCheckboxComponent.ctorParameters = function () { return []; };
    LibraryCheckboxComponent.propDecorators = {
        name: [{ type: Input }]
    };
    return LibraryCheckboxComponent;
}());
export { LibraryCheckboxComponent };
if (false) {
    /** @type {?} */
    LibraryCheckboxComponent.prototype.name;
}
