// abz-components-library/fesm2015/abz-components-library.js
import { Component, Input, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * Examples:  <example-url>https://stackblitz.com/edit/angular-sw5vxm?embed=1&file=src/app/app.module.ts</example-url>
 */
class LibraryButtonComponent {
    constructor() {
        this.name = 'default';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
LibraryButtonComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-button',
                template: `<button type="button">Click Me!</button>
`,
                styles: []
            },] },
];
LibraryButtonComponent.ctorParameters = () => [];
LibraryButtonComponent.propDecorators = {
    name: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * Examples:  <example-url>https://stackblitz.com/edit/angular-sw5vxm?embed=1&file=src/app/app.module.ts</example-url>
 */
class LibraryCheckboxComponent {
    constructor() {
        this.name = 'default';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
LibraryCheckboxComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-checkbox',
                template: `<input type="checkbox" name="vehicle1" value="Bike"> I have a bike<br>
<input type="checkbox" name="vehicle2" value="Car"> I have a car<br>
<input type="checkbox" name="vehicle3" value="Boat" checked> I have a boat<br><br>
`,
                styles: []
            },] },
];
LibraryCheckboxComponent.ctorParameters = () => [];
LibraryCheckboxComponent.propDecorators = {
    name: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class ComponentsLibraryModule {
}
ComponentsLibraryModule.decorators = [
    { type: NgModule, args: [{
                imports: [],
                declarations: [LibraryButtonComponent, LibraryCheckboxComponent],
                exports: [LibraryButtonComponent, LibraryCheckboxComponent]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { LibraryButtonComponent, ComponentsLibraryModule, LibraryCheckboxComponent as ɵa };
