// abz-components-library/fesm2015/components-library.js
import { Injectable, NgModule, Component, defineInjectable } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class ComponentsLibraryService {
    constructor() { }
}
ComponentsLibraryService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
ComponentsLibraryService.ctorParameters = () => [];
/** @nocollapse */ ComponentsLibraryService.ngInjectableDef = defineInjectable({ factory: function ComponentsLibraryService_Factory() { return new ComponentsLibraryService(); }, token: ComponentsLibraryService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class ComponentsLibraryComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
ComponentsLibraryComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-components-library',
                template: `
    <p>
      components-library works!
    </p>
  `,
                styles: []
            },] },
];
ComponentsLibraryComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class ComponentsLibraryModule {
}
ComponentsLibraryModule.decorators = [
    { type: NgModule, args: [{
                imports: [],
                declarations: [ComponentsLibraryComponent],
                exports: [ComponentsLibraryComponent]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { ComponentsLibraryService, ComponentsLibraryComponent, ComponentsLibraryModule };
